elastic/kibana
# [Fleet] update getPackageInfo to handle uploaded packages

`getPackageInfo` is the handler for the route `/api/fleet/epm/packages/{pkg}`.  Before this change it always reached out to the registry to get the package.  For example if you uploaded a package called `apache-0.1.4` and  visited `/api/fleet/epm/packages/apache-0.1.4` the registry package would be returned and not the one you uploaded.  This updates to check the install source, if installed, and fetch from that source.

- update `getPackageInfo` to look at install source before looking for the package.  If an uploaded package doesn't exist in memory, it will throw an error. This will be updated once the package storage is available and it would look there next before throwing an error, probably a 404.

- update the .zip and .tar.gz fixtures for apache-0.1.4 package to differentiate itself from the one in the registry to make testing easier.  the description in the manifest was changed and the README.md in /docs to say "Apache Uploaded Test Integration".  This is compared in an integration test and you can easily see the different when navigating to the package details in the UI

- introduces `getPackageFromSource` which can be updated once the package storage is available.

This change is needed for https://github.com/elastic/kibana/issues/83312 and https://github.com/elastic/kibana/issues/83311


Test:

1. Upload a package 
`curl -X POST -u elastic:changeme http://localhost:5603/ssg/api/fleet/epm/packages --data-binary @dev/elastic/kibana/x-pack/test/fleet_api_integration/apis/fixtures/direct_upload_packages/apache_0.1.4.tar.gz -H 'kbn-xsrf: xyz' -H 'Content-Type: application/gzip'`
2. Visit the route and see that the uploaded package information is being returned
<img width="554" alt="Screen Shot 2020-11-19 at 2 56 34 PM" src="https://user-images.githubusercontent.com/1676003/99717474-6fcb4780-2a77-11eb-9fc4-f9d1ab6ecb40.png">

3.  Uninstall apache-0.1.4
4. Visit the route again and see the registry version is returned
<img width="651" alt="Screen Shot 2020-11-19 at 2 55 55 PM" src="https://user-images.githubusercontent.com/1676003/99717403-5b874a80-2a77-11eb-9410-52cba8f84b58.png">
